packages/boxel-motion-demo-app/app/controllers/motion-study.ts

```diff
@@ -1,14 +1,170 @@
+    });
+
+    let cardContentSprites = changeset.spritesFor({
+      role: 'card-content',
+      type: SpriteType.Inserted,
+    });
+
+    let fadeOutClosingCardContent = {
+      sprites: removedCardContentSprites,
+      properties: {
+        opacity: { to: 0 },
+      },
+      timing: {
+        behavior: new TweenBehavior(),
+        duration: fadeDuration,
+      },
+    };
+
+    let moveClosingCardContentToForeground = {
+      sprites: removedCardContentSprites,
+      properties: {
+        zIndex: 2,
+      },
+      timing: {
+        behavior: new StaticBehavior(),
+        duration: fadeDuration,
+      },
+    };
+
+    let keepClosingCardContentHidden = {
+      sprites: removedCardContentSprites,
+      properties: {
+        opacity: 0,
+      },
+      timing: {
+        behavior: new StaticBehavior(),
+        duration: resizeAndMoveDuration,
+      },
+    };
+
+    let moveAllCardsToMidground = {
+      sprites: cardSprites,
+      properties: {
+        zIndex: 1,
+      },
+      timing: {
+        behavior: new StaticBehavior(),
+        duration: fadeDuration,
+      },
+    };
+
+    let moveNonAnimatingCardsToBackground = {
+      sprites: new Set(nonAnimatingCardSprites),
+      properties: {
+        zIndex: 0,
+      },
+      timing: {
+        behavior: new StaticBehavior(),
+        duration: resizeAndMoveDuration,
+      },
+    };
+
+    let resizeAnimatingCard = {
+      sprites: cardSprites,
+      properties: {
+        translateX: {},
+        translateY: {},
+        width: {},
+        height: {},
+      },
+      timing: {
+        // TODO convert to SpringBehavior when its duration can be referenced by other animations
+        behavior: new TweenBehavior({
+          easing: easeInAndOut,
+        }),
+        duration: resizeAndMoveDuration,
+      },
+    };
+
+    let keepCardsBeingRemovedUntilOpeningCompletes = {
+      sprites: removedCardSprites,
+      properties: {},
+      timing: {
+        behavior: new WaitBehavior(),
+        duration: fadeDuration,
+      },
+    };
+
+    let fadeInOpeningCardContent = {
+      sprites: cardContentSprites,
+      properties: {
+        opacity: { from: 0 },
+      },
+      timing: {
+        behavior: new TweenBehavior(),
+        duration: fadeDuration,
+      },
+    };
+
+    return {
+      timeline: {
+        type: 'sequence',
+        animations: [
+          {
+            type: 'parallel',
+            animations: [
+              moveClosingCardContentToForeground,
+              fadeOutClosingCardContent,
+            ],
+          },
+          {
+            type: 'parallel',
+            animations: [
+              keepClosingCardContentHidden,
+              moveAllCardsToMidground,
+              moveNonAnimatingCardsToBackground,
+              resizeAnimatingCard,
+              keepCardsBeingRemovedUntilOpeningCompletes,
+            ],
+          },
+          fadeInOpeningCardContent,
+        ],
+      },
+    };
```

Comment:
Thoughts re these extracted sub-animations, @Aierie and @nickschot? It feels helpful to me as a n00b but is it just annoying? 🤔

Reply:
No that's fine! It'll help us with refining a final API later. As you saw it can get quite confusing if you use a big definition (even though it's quite a bit better than the situation we had before this declarative API).